web/packages/shared/components/Select/types.ts

```diff
@@ -39,7 +39,7 @@ export type Props = {
   autoFocus?: boolean;
   label?: string;
   placeholder?: string;
-  options: Option<any, any>[] | GroupOption[];
+  options?: Option<any, any>[] | GroupOption[];
```

Comment:
Why does it need to be optional? How does `Select` work if no options are passed to it? As in, what is its purpose if it doesn't receive any options?

Reply:
Exactly my thought and I honestly do not know why it was explicitly omitted where it is currently defined: https://github.com/gravitational/teleport/blob/224b1414d69983e54af2a830997ae107d06eb641/web/packages/shared/components/Select/types.ts#L69 This PR now removes the omission, but I do not understand original intention for what it seems like "obvious" so I left it be optional.

Reply:
looking at react-select docs, it's actually a required field but there is a component that uses it in a way that we don't render the dropdown at all (so options didn't need to be passed)

<img width="1311" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/13c0f4dc-48c0-4b5c-aeb7-19f02037b646">

Reply:
Thanks @kimlisa for deeper look. I was assuming the same, since it's a creatable component, somewhere, it must have been used without the predefined option.